crates/weaver_resolved_schema/src/error.rs

```diff
@@ -0,0 +1,49 @@
+// SPDX-License-Identifier: Apache-2.0
+
+//! Error types and utilities.
+
+use crate::attribute::AttributeRef;
+use crate::error::Error::CompoundError;
+
+/// Errors emitted by this crate.
+#[derive(thiserror::Error, Debug, Clone)]
+pub enum Error {
+    /// Attribute reference not found in the catalog.
+    #[error("Attribute reference {attr_ref} (group: {group_id}) not found in the catalog")]
+    AttributeNotFound {
+        /// Group id.
+        group_id: String,
+        /// Attribute reference.
+        attr_ref: AttributeRef,
+    },
+
+    /// A generic container for multiple errors.
+    #[error("Errors:\n{0:#?}")]
+    CompoundError(Vec<Error>),
+}
+
+/// Handles a list of errors and returns a compound error if the list is not
+/// empty or () if the list is empty.
+pub fn handle_errors(errors: Vec<Error>) -> Result<(), Error> {
```

Comment:
Maybe we shouldn't carry this everywhere. I wonder if we could abstract it in some way...

Reply:
Yes, I agree; I was thinking the same thing. I need to explore ways to unify that in a future PR.